tests/test_baseline.py

```diff
@@ -0,0 +1,275 @@
+        )
+        for embedding_sliced, embedding_ct in zip(
+            embeddings_sliced.values(), self.embeddings_ct.values()
+        ):
+            assert torch.all(
+                torch.abs(embedding_sliced - embedding_ct[::2, ...]) < 1e-5
+            )
+
+        # Test for not centered
+        embeddings_sliced, _ = get_audio_embedding(
+            audio=audio_sliced,
+            model=self.model,
+            hop_size=256,
+            batch_size=512,
+            center=False,
+        )
+        for embedding_sliced, embedding_not_ct in zip(
+            embeddings_sliced.values(), self.embeddings_not_ct.values()
+        ):
+            assert torch.all(
+                torch.abs(embedding_sliced - embedding_not_ct[::2, ...]) < 1e-5
+            )
+
+    def test_embeddings_shape(self):
+        # Test the embeddings shape for centered and not centered.
+        # The embeddings size in these two cases are different by the codes
+        # logic.
+        for size, embedding in self.embeddings_not_ct.items():
+            assert embedding.shape == (64, 96000 // 256, int(size))
+
+        for size, embedding in self.embeddings_ct.items():
+            assert embedding.shape == (64, (4096 // 2 + 96000) // 256, int(size))
+
+    def test_embeddings_nan(self):
+        # Test for null values in the embeddings.
+        for embeddings in [self.embeddings_ct, self.embeddings_not_ct]:
+            for size, embedding in embeddings.items():
+                assert not torch.any(torch.isnan(embedding))
+
+    def test_embeddings_type(self):
+        # Test the data type of the embeddings.
+        for embeddings in [self.embeddings_ct, self.embeddings_not_ct]:
+            for size, embedding in embeddings.items():
+                if size != 20:
+                    assert embedding.dtype == torch.float32
+                else:
+                    assert embedding.dtype == torch.int8
+
+    def test_timestamps_begin(self):
+        # Test the beginning of the time stamp in case of centered and not
+        # centered.
+        assert self.ts_ct[0] == 0
+        assert (
+            torch.abs(self.ts_not_ct[0] - int(4096 // 2) / input_sample_rate()) < 1e-5
+        )
+
+    def test_timestamps_spacing(self):
+        # Test the spacing between the time stamp in case of centered and not
+        # centered.
+        assert torch.all(torch.abs(torch.diff(self.ts_ct) - self.ts_ct[1]) < 1e-5)
+        assert torch.all(
+            torch.abs(
+                torch.diff(self.ts_not_ct) - (self.ts_not_ct[2] - self.ts_not_ct[1])
+            )
+            < 1e-5
+        )
+
+    def test_timestamps_end(self):
+        # Test the end of the timestamp. This should technically not exceed the
+        # time of the audio. However it is hapenning in our case though?
+        assert torch.abs(self.ts_ct[-1] - 96000 / input_sample_rate()) < 1e-5
+        assert torch.abs(self.ts_not_ct[-1] - 96000 / input_sample_rate()) < 1e-5
```

Comment:
It depends on how you want to define the centring behaviour. Either way, I made a suggestion to fix this in the comments for #2.